src/v/archival/ntp_archiver_service.cc

```diff
@@ -1721,6 +1725,15 @@ ntp_archiver::find_reupload_candidate(manifest_scanner_t scanner) {
             vlog(_rtclog.warn, "Failed to make upload candidate");
             co_return std::nullopt;
         }
+        if (candidate.candidate.content_length != run->meta.size_bytes) {
```

Comment:
Is it guaranteed that the value of `run->meta.committed_offset` used as target end will be aligned to the manifest? It seems like it is because the value comes from the manifest itself, but if it is not then the `align_end_offset...` in the collector might end up cutting back the content length to align failing this condition.

Reply:
it's not clear why the CI fails, the segments shouldn't overlap, I added extra logging to check